posthog/temporal/workflows/s3_batch_export.py

```diff
@@ -96,35 +97,53 @@ async def insert_into_s3_activity(inputs: S3InsertInputs):
             aws_access_key_id=inputs.aws_access_key_id,
             aws_secret_access_key=inputs.aws_secret_access_key,
         )
-        multipart_response = s3_client.create_multipart_upload(Bucket=inputs.bucket_name, Key=key)
-        upload_id = multipart_response["UploadId"]
+        details = activity.info().heartbeat_details
+
+        parts: List[CompletedPartTypeDef] = []
+
+        if len(details) == 2:
```

Comment:
```suggestion
        if len(details) == 4:
```

Reply:
Yeah, this is precisely the "if we decide we need to change these values we'll have issues" you mentioned

Reply:
I'll have #16445 ready tomorrow to glob up all this complexity, so at least is in one place...

Reply:
And perhaps use a tuple here, so that the value is always 1